### ISSUE TITLE
Implement simple VTK viewer

### ISSUE THREAD

**Issue by astrofrog** [glue-viz/glue#589]
While 3-d selection is going to be hard to do, it should be possible to develop a simple 3-D data viewer. I've started working on this, though it's low priority so may be a while until it's done. Here's a teaser:

![screen shot 2015-03-15 at 10 54 51 pm](https://cloud.githubusercontent.com/assets/314716/6658348/cf853634-cb66-11e4-8c9c-5151049b04ce.png)

Of course, we need to hope VTK eventually have Python 3 support...

(this is a good use case for developing a new Qt widget, so taking notes along the way)

**Comment by ChrisBeaumont** [comment]
Cool! I had a demo of something similar on an ancient branch. I don't have that computer on me but I'll upload the branch when I get the chance (however, it looks like you've already reproduced most of the functionality I implemented, so it might not be too useful).

We should also take another look at vispy http://vispy.org/ as an option for 3D stuff

**Comment by astrofrog** [comment]
@ChrisBeaumont - I'll definitely look into vispy! I could try and implement both as possible plugin widgets since it will be a good exercise anyway :)

**Comment by astrofrog** [comment]
@ChrisBeaumont - side note: just tried vispy for 2-d plots (scatter plots) and it's blazing fast! Only issue is, it doesn't show axes, which is a big issue. Glumpu (http://glumpy.github.io/gallery.html) also looks promising and does have some example 3-d plots with axes so maybe 2-d with axes is also possible.

**Comment by astrofrog** [comment]
While working on #619 (writing custom Qt clients), I also worked in parallel on testing this out with the VTK cube viewer which I've now put here:

https://github.com/astrofrog/cube-viewer

It's intended to be used as standalone, and there is a thin wrapper that makes it usable as a glue data viewer:

https://github.com/astrofrog/cube-viewer/blob/master/cube_viewer/glue_viewer.py

To register it as a glue viewer, one should install the package, then add the following to ``config.py``:

```python
from glue.config import qt_client
qt_client.lazy_add('cube_viewer.glue_plugin')
```

Note that this doesn't yet support displaying subsets - that will be the next step.

**Comment by astrofrog** [comment]
I'm going to try using vispy instead: http://vispy.org/examples/basics/scene/volume.html - supports Python 3 and seems nicer API-wise!

**Comment by astrofrog** [comment]
This is being developed in https://github.com/glue-viz/glue-3d-viewer. We can close this issue once there is a release.

**Comment by astrofrog** [comment]
I'm going to close this since we now have 3D viewers based on VisPy: https://github.com/glue-viz/glue-vispy-viewers
